### DISCUSSION TITLE
PlugnMeet - Error: could not establish pc connection

### DISCUSSION THREAD

**Discussion by Resul0316** [mynaparrot/plugNmeet-server#434] [Q&A]
I have a problem when i use plugnMeet. When i try to enter a video call after clicking the microphone it gives me this error:  Error: could not establish pc connection.

I have tried the find from internet or plugnMeet forum if the problem is solved but there is not a concrete solution. In documentation it suggest me as:  --config "/app/livekit.yaml" --dev --node-ip YOUR_IP_HERE  
-- https://www.plugnmeet.org/docs/developer-guide/setup-development/

However, even though i do as it is indicated in documentation the problem persist. Does anyone has a idea what can i do to resolve the problem?

I use ubuntu version 22.04.

I tried to enter the main page of the video call. In my code when i comment the part 
await this._room.connect(this.url, this.token); 
and try to enter it makes me enter to the main page but as soon as i enter i have to uncomment this part. In this way it works and after entering first time i can enter always without problem. However, it is not a correct way. First time without commenting this part how it is possible to manage this problem.

**Comment by jibon57** [comment]
Did you replace the value of `YOUR_IP_HERE` with your machine IP address?

**Reply by Resul0316** [reply]
Yes i followed every step as indicated in documentation. This gives my ip address as you commented in another discussion: ip route get 8.8.8.8 | awk -F "src " 'NR==1{split($2,a," ");print a[1]}' 

I changed it as : --config "/app/livekit.yaml" --dev --node-ip 192.168.5.112

However, the problem still persist.

**Reply by jibon57** [reply]
After change need to restart livekit. Also can you provide the logs of livekit? Need to check if livekit used that provided IP or not.

**Reply by Resul0316** [reply]
I am adding here the logs:

`server-livekit-1          | 2024-02-27T12:02:27.148Z    DEBUG   livekit routing/redisrouter.go:445      starting redisWorker    {"nodeID": "ND_63za8hYYvTqL"}
server-livekit-1          | 2024-02-27T12:02:27.152Z    INFO    livekit service/server.go:242   starting LiveKit server {"portHttp": 7880, "nodeID": "ND_63za8hYYvTqL", "nodeIP": "192.168.5.112", "version": "1.5.2", "rtc.portTCP": 7881, "rtc.portUDP": {"Start":7882,"End":0}}
server-livekit-1          | 2024-02-27T12:02:27.152Z    DEBUG   livekit service/signal.go:115   starting relay signal server    {"topic": "ND_63za8hYYvTqL"}`
`server-livekit-1          | 2024-02-27T12:02:27.053Z    INFO    livekit redis/redis.go:133      connecting to redis     {"simple": true, "addr": "redis:6379"}`


if we are talking about any other log please let me know. Thanks.

**Reply by jibon57** [reply]
Log looks OK, how about during join? Also, which browser you're trying from?

**Reply by Resul0316** [reply]
I write api_key and secret and proceed to submit. And it takes me to the  localhost:3000/?access_token=... page. There is a modal and i click to the microphone. And if i comment await this._room.connect(this.url, this.token); this code and click and then after entering uncomment this code everything works well. However, this is not correct way of proceding. If i don't comment the code after 10 seconds I take the indicated error on console. On network I don't have any error (means not making an api call). 

ConnectLivekit.ts:43 Error: could not establish pc connection
    at livekit-client.esm.mjs:16953:1

I use ubuntu version 22.04.
browser:  "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36"
browser: "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:122.0) Gecko/20100101 Firefox/122.0"

However, the problem also happens on windows.

**Reply by jibon57** [reply]
Please provide the log of livekit that provided during join. From my side I can't see any problem. This error happen when livekit connection being blocked. You may try this option too: https://github.com/livekit/livekit/blob/7649e4ffab39143ba7ab41919e955f2cf4064a89/config-sample.yaml#L96

**Reply by Resul0316** [reply]
it is a bit long but here is the all livekit log during login:

server-livekit-1          | 2024-02-27T13:24:04.469Z    INFO    livekit service/roomallocator.go:138    selected node for room  {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "selectedNodeID": "ND_Ew9838fXcbQA"}
server-livekit-1          | 2024-02-27T13:24:04.469Z    DEBUG   livekit routing/signal.go:100   starting signal connection      {"room": "room01", "reqNodeID": "ND_Ew9838fXcbQA", "participant": "", "connID": "CO_RWgKjypDSQ7v"}
server-livekit-1          | 2024-02-27T13:24:04.470Z    DEBUG   livekit service/signal.go:168   signal stream closed    {"room": "room01", "participant": "", "connID": "CO_RWgKjypDSQ7v", "error": null}
server-livekit-1          | 2024-02-27T13:24:04.470Z    DEBUG   livekit routing/signal.go:136   signal stream closed    {"room": "room01", "reqNodeID": "ND_Ew9838fXcbQA", "participant": "", "connID": "CO_RWgKjypDSQ7v", "error": "stream closed"}
server-livekit-1          | 2024-02-27T13:24:04.471Z    INFO    livekit.api     service/twirp.go:124    API RoomService.CreateRoom     

server-plugnmeet-api-1    | 13:24:04 | 200 |   94.738167ms | 172.18.0.1 | POST | /auth/room/create | -
server-livekit-1          | 2024-02-27T13:24:04.532Z    DEBUG   livekit.room    service/roommanager.go:778      updating room   {"room": "room01", "roomID": "RM_hhCFSzd3Arqn"}
server-plugnmeet-api-1    | 13:24:04 | 204 |      51.115µs | 172.18.0.1 | OPTIONS | /auth/room/getJoinToken | -
server-livekit-1          | 2024-02-27T13:24:04.533Z    DEBUG   livekit.psrpc.Room.UpdateRoomMetadata   psrpc/logging.go:69  

server-livekit-1          | 2024-02-27T13:24:04.533Z    INFO    livekit.api     service/twirp.go:124    API RoomService.UpdateRoomMetadata      {"service": "RoomService", "method": "UpdateRoomMetadata", "room": "room01", "size": 2108, "duration": "54.092235ms", "status": "200"}
server-livekit-1          | 2024-02-27T13:24:04.581Z    DEBUG   livekit.room    service/roommanager.go:761      api send data   {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "size": 2199}
server-livekit-1          | 2024-02-27T13:24:04.582Z    DEBUG   livekit.psrpc.Room.SendData     psrpc/logging.go:69     client response {"topic": ["room01"], "request": {"room": "room01", "data": "CAESJGQ1YjE2Nzk4LTQzNjItNDQwOC04MzBmLTFhNDIwZDFkMGJhNDLsEAgZGh9UdWUsIDI3IEZlYiAyMDI0IA... (2.15kB)"}, "response": {}, "duration": "1.21382ms"}
server-livekit-1          | 2024-02-27T13:24:04.582Z    INFO    livekit.api     service/twirp.go:124    API RoomService.SendData        {"service": "RoomService", "method": "SendData", "room": "room01", "size": 2199, "duration": "1.505709ms", "status": "200"}
server-livekit-1          | 2024-02-27T13:24:04.604Z    INFO    livekit.webhook webhook/url_notifier.go:88      sent webhook 

server-plugnmeet-api-1    | 13:24:04 | 200 |  132.764572ms | 172.18.0.5 | POST | /webhook | -
server-livekit-1          | 2024-02-27T13:24:04.648Z    INFO    livekit.api     service/twirp.go:124    API RoomService.ListParticipants        {"service": "RoomService", "method": "ListParticipants", "room": "room01", "duration": "494.824µs", "status": "200"}
server-plugnmeet-api-1    | 13:24:04 | 200 |   70.474801ms | 172.18.0.1 | POST | /auth/room/getJoinToken | -
server-plugnmeet-api-1    | 13:24:06 | 204 |      39.253µs | 172.18.0.1 | OPTIONS | /api/verifyToken | -
server-livekit-1          | 2024-02-27T13:24:06.697Z    INFO    livekit.api     service/twirp.go:124    API RoomService.ListRooms       {"service": "RoomService", "method": "ListRooms", "room": ["room01"], "duration": "272.919µs", "status": "200"}
server-plugnmeet-api-1    | 13:24:06 | 200 |   46.891176ms | 172.18.0.1 | POST | /api/verifyToken | -
server-livekit-1          | 2024-02-27T13:24:09.275Z    DEBUG   livekit routing/signal.go:100   starting signal connection      {"room": "room01", "reqNodeID": "ND_Ew9838fXcbQA", "participant": "1709032784366", "connID": "CO_Tg8sQxaDM7tp"}
server-livekit-1          | 2024-02-27T13:24:09.276Z    DEBUG   livekit service/roommanager.go:338      starting RTC session    {"room": "room01", "nodeID": "ND_Ew9838fXcbQA", "participant": "1709032784366", "clientInfo": {"sdk": 1, "version": "1.15.10", "protocol": 11, "os": "Linux", "deviceModel": "Other", "browser": "Chrome", "browserVersion": "122.0.0", "address": "172.18.0.1"}, "reconnect": false, "reconnectReason": "RR_UNKNOWN", "adaptiveStream": true, "numParticipants": 0}
server-livekit-1          | 2024-02-27T13:24:09.277Z    DEBUG   livekit.room    rtc/room.go:405 new participant joined  {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "pID": "PA_i4nkHXxCEPXP", "participant": "1709032784366", "clientInfo": {"sdk": 1, "version": "1.15.10", "protocol": 11, "os": "Linux", "deviceModel": "Other", "browser": "Chrome", "browserVersion": "122.0.0", "address": "172.18.0.1"}, "options": {"AutoSubscribe":true}, "numParticipants": 0}
server-livekit-1          | 2024-02-27T13:24:09.277Z    DEBUG   livekit rtc/participant.go:1247 updating participant state      {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "participant": "1709032784366", "pID": "PA_i4nkHXxCEPXP", "remote": false, "state": "JOINED"}
server-livekit-1          | 2024-02-27T13:24:09.277Z    DEBUG   livekit rtc/participant.go:879  SetMigrateState {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "participant": "1709032784366", "pID": "PA_i4nkHXxCEPXP", "remote": false, "state": "MIGRATE_STATE_COMPLETE"}
server-livekit-1          | 2024-02-27T13:24:09.278Z    DEBUG   livekit.transport       rtc/transport.go:579    ice gathering state change      {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "participant": "1709032784366", "pID": "PA_i4nkHXxCEPXP", "remote": false, "transport": "SUBSCRIBER", "state": "gathering"}
server-livekit-1          | 2024-02-27T13:24:09.278Z    DEBUG   livekit.sub     rtc/participant.go:1291 sending offer   {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "participant": "1709032784366", "pID": "PA_i4nkHXxCEPXP", "remote": false, "transport": "SUBSCRIBER"}
server-livekit-1          | 2024-02-27T13:24:09.278Z    INFO    livekit service/rtcservice.go:309       new client WS connected {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "participant": "1709032784366", "pID": "PA_i4nkHXxCEPXP", "remote": false, "connID": "CO_Tg8sQxaDM7tp", "reconnect": false, "reconnectReason": "RR_UNKNOWN", "adaptiveStream": true, "selectedNodeID": "ND_Ew9838fXcbQA"}
server-livekit-1          | 2024-02-27T13:24:09.279Z    DEBUG   livekit service/rtcservice.go:348       sending offer   {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "participant": "1709032784366", "pID": "PA_i4nkHXxCEPXP", "remote": false, "offer": {"Offer":{"type":"offer","sdp":"v=0\r\no=- 2181197261082106980 1709040249 IN IP4 0.0.0.0\r\ns=-\r\nt=0 0\r\na=fingerprint:sha-256 AA:C6:26:B5:2E:A0:70:EE:BF:A0:E6:B3:04:C1:B6:6D:2A:15:A3:EC:38:62:85:37:7C:D5:1F:AD:03:A7:91:47\r\na=extmap-allow-mixed\r\na=group:BUNDLE 0\r\nm=application 9 UDP/DTLS/SCTP webrtc-datachannel\r\nc=IN IP4 0.0.0.0\r\na=setup:actpass\r\na=mid:0\r\na=sendrecv\r\na=sctp-port:5000\r\na=ice-ufrag:HxSUgVEMixWyxLYc\r\na=ice-pwd:hgYYDzcpiqEkFRRbADghQqHnlFpmpLmd\r\n"}}}
server-livekit-1          | 2024-02-27T13:24:09.336Z    DEBUG   livekit service/rtcservice.go:428       received answer {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "participant": "1709032784366", "pID": "PA_i4nkHXxCEPXP", "remote": false, "answer": {"Answer":{"type":"answer","sdp":"v=0\r\no=- 3298212506531161290 2 IN IP4 127.0.0.1\r\ns=-\r\nt=0 0\r\na=extmap-allow-mixed\r\na=group:BUNDLE 0\r\nm=application 9 UDP/DTLS/SCTP webrtc-datachannel\r\nc=IN IP4 0.0.0.0\r\na=setup:active\r\na=mid:0\r\na=ice-ufrag:E1Lh\r\na=ice-pwd:+RMwhz+PeMjRRLwkiYbuCM2X\r\na=fingerprint:sha-256 83:5E:3F:E8:07:98:D2:C3:D4:97:B3:48:1C:40:95:FB:35:C3:E6:88:90:16:0B:CC:AA:EA:D5:8A:DB:EC:0D:AF\r\na=ice-options:trickle\r\na=sctp-port:5000\r\n"}}}
server-livekit-1          | 2024-02-27T13:24:09.339Z    DEBUG   livekit.sub     rtc/participant.go:604  received answer {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "participant": "1709032784366", "pID": "PA_i4nkHXxCEPXP", "remote": false, "transport": "SUBSCRIBER"}
server-livekit-1          | 2024-02-27T13:24:09.343Z    DEBUG   livekit.transport       rtc/transportmanager.go:450     ignoring mDNS candidate {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "participant": "1709032784366", "pID": "PA_i4nkHXxCEPXP", "remote": false, "candidate": "2620293423 1 udp 2113937151 c376aa06-fdb3-48b5-8a3e-beb6d1e7cb18.local 53854 typ host generation 0 ufrag E1Lh network-cost 999", "target": "SUBSCRIBER"}
server-livekit-1          | 2024-02-27T13:24:09.344Z    DEBUG   livekit.transport       rtc/transport.go:619    ice connection state change     {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "participant": "1709032784366", "pID": "PA_i4nkHXxCEPXP", "remote": false, "transport": "SUBSCRIBER", "state": "checking"}
server-livekit-1          | 2024-02-27T13:24:09.344Z    DEBUG   livekit.transport       rtc/transport.go:643    peer connection state change    {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "participant": "1709032784366", "pID": "PA_i4nkHXxCEPXP", "remote": false, "transport": "SUBSCRIBER", "state": "connecting"}
server-livekit-1          | 2024-02-27T13:24:09.377Z    DEBUG   livekit.transport       rtc/transport.go:579    ice gathering state change      {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "participant": "1709032784366", "pID": "PA_i4nkHXxCEPXP", "remote": false, "transport": "SUBSCRIBER", "state": "complete"}
server-livekit-1          | 2024-02-27T13:24:24.335Z    INFO    livekit service/rtcservice.go:269       finishing WS connection {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "participant": "1709032784366", "pID": "PA_i4nkHXxCEPXP", "remote": false, "connID": "CO_Tg8sQxaDM7tp", "closedByClient": true}
server-livekit-1          | 2024-02-27T13:24:24.335Z    DEBUG   livekit service/rtcservice.go:335       nothing to read from response source    {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "participant": "1709032784366", "pID": "PA_i4nkHXxCEPXP", "remote": false, "connID": "CO_Tg8sQxaDM7tp"}
server-livekit-1          | 2024-02-27T13:24:24.337Z    DEBUG   livekit service/signal.go:168   signal stream closed    {"room": "room01", "participant": "1709032784366", "connID": "CO_Tg8sQxaDM7tp", "error": null}
server-livekit-1          | 2024-02-27T13:24:24.337Z    DEBUG   livekit.transport       rtc/transportmanager.go:801     signal source valid     {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "participant": "1709032784366", "pID": "PA_i4nkHXxCEPXP", "remote": false, "valid": false}
server-livekit-1          | 2024-02-27T13:24:24.337Z    INFO    livekit rtc/participant.go:741  participant closing     {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "participant": "1709032784366", "pID": "PA_i4nkHXxCEPXP", "remote": false, "sendLeave": false, "reason": "JOIN_FAILED", "isExpectedToResume": false}
server-livekit-1          | 2024-02-27T13:24:24.337Z    DEBUG   livekit rtc/participant.go:1247 updating participant state      {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "participant": "1709032784366", "pID": "PA_i4nkHXxCEPXP", "remote": false, "state": "DISCONNECTED"}
server-livekit-1          | 2024-02-27T13:24:24.337Z    INFO    livekit rtc/room.go:545 removing participant without connection {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "participant": "1709032784366", "pID": "PA_i4nkHXxCEPXP", "remote": false, "subscriberCandidates": ["[local] udp4 host 172.18.0.5:7882", "[local] tcp4 host 172.18.0.5:7881", "[local] udp4 srflx 2.42.32.137:51376 related 0.0.0.0:52550", "[remote] udp4 srflx 2.42.32.137:53854 related 0.0.0.0:0"], "connectionType": "unknown", "reason": "STATE_DISCONNECTED"}
server-livekit-1          | 2024-02-27T13:24:24.337Z    DEBUG   livekit routing/signal.go:136   signal stream closed    {"room": "room01", "reqNodeID": "ND_Ew9838fXcbQA", "participant": "1709032784366", "connID": "CO_Tg8sQxaDM7tp", "error": "stream closed"}
server-livekit-1          | 2024-02-27T13:24:24.338Z    DEBUG   livekit service/roommanager.go:590      RTC session finishing   {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "participant": "1709032784366", "pID": "PA_i4nkHXxCEPXP", "remote": false, "connID": "CO_Tg8sQxaDM7tp"}
server-livekit-1          | 2024-02-27T13:24:24.338Z    DEBUG   livekit.transport       rtc/transport.go:1420   leaving events processor        {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "participant": "1709032784366", "pID": "PA_i4nkHXxCEPXP", "remote": false, "transport": "SUBSCRIBER"}
server-livekit-1          | 2024-02-27T13:24:24.338Z    DEBUG   livekit.transport       rtc/transport.go:643    peer connection state change    {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "participant": "1709032784366", "pID": "PA_i4nkHXxCEPXP", "remote": false, "transport": "PUBLISHER", "state": "closed"}
server-livekit-1          | 2024-02-27T13:24:24.338Z    DEBUG   livekit.transport       rtc/transport.go:1420   leaving events processor        {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "participant": "1709032784366", "pID": "PA_i4nkHXxCEPXP", "remote": false, "transport": "PUBLISHER"}
server-livekit-1          | 2024-02-27T13:24:24.338Z    DEBUG   livekit.transport       rtc/transport.go:579    ice gathering state change      {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "participant": "1709032784366", "pID": "PA_i4nkHXxCEPXP", "remote": false, "transport": "SUBSCRIBER", "state": "closed"}
server-livekit-1          | 2024-02-27T13:24:24.338Z    DEBUG   livekit.transport       rtc/transport.go:619    ice connection state change     {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "participant": "1709032784366", "pID": "PA_i4nkHXxCEPXP", "remote": false, "transport": "SUBSCRIBER", "state": "closed"}
server-livekit-1          | 2024-02-27T13:24:24.338Z    DEBUG   livekit.transport       rtc/transport.go:643    peer connection state change    {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "participant": "1709032784366", "pID": "PA_i4nkHXxCEPXP", "remote": false, "transport": "SUBSCRIBER", "state": "closed"}
server-livekit-1          | 2024-02-27T13:24:44.470Z    INFO    livekit.room    rtc/room.go:742 closing room    {"room": "room01", "roomID": "RM_hhCFSzd3Arqn"}
server-livekit-1          | 2024-02-27T13:24:44.470Z    INFO    livekit service/roommanager.go:145      deleting room state     {"room": "room01"}
server-livekit-1          | 2024-02-27T13:24:44.471Z    INFO    livekit.room    service/roommanager.go:552      room closed     {"room": "room01", "roomID": "RM_hhCFSzd3Arqn"}
server-livekit-1          | 2024-02-27T13:24:44.481Z    INFO    livekit.webhook webhook/url_notifier.go:88      sent webhook

**Reply by jibon57** [reply]
```
 server-livekit-1 | 2024-02-27T13:24:24.337Z INFO livekit rtc/room.go:545 removing participant without connection {"room": "room01", "roomID": "RM_hhCFSzd3Arqn", "participant": "1709032784366", "pID": "PA_i4nkHXxCEPXP", "remote": false, "subscriberCandidates": ["[local] udp4 host 172.18.0.5:7882", "[local] tcp4 host 172.18.0.5:7881", "[local] udp4 srflx 2.42.32.137:51376 related 0.0.0.0:52550", "[remote] udp4 srflx 2.42.32.137:53854 related 0.0.0.0:0"], "connectionType": "unknown", "reason": "STATE_DISCONNECTED"}
```

Not sure why it's trying to connect with your external IP `2.42.32.137:51376`. Are you trying to join from the same machine were you have setup the dev environment? Not any virtual machine?

**Reply by Resul0316** [reply]
I am not sure if you can replicate the problem i face but maybe you can try in incognito. Or cancel all cookies from your current browser and try to replicate.

**Reply by Resul0316** [reply]
I am trying to enter from same machine, not virtual environment. By the way, not only me also my colleagues are facing the same problem.

**Reply by jibon57** [reply]
As it's livekit related problem & from my end I can not reproduce it. You may join slack & write here: http://livekit.io/join-slack

**Reply by Resul0316** [reply]
Okay I will explain the problem in livekit helpdesk.
